README.md

```diff
@@ -1,3 +1,134 @@
+    ```$xslt
+        <groupId>com.google.sps</groupId>
+        <artifactId>INSERT_PROJECT_ID_HERE</artifactId>
+        <version>1</version>
+        <packaging>war</packaging>
+    ```
+Replace the `artifactId` with your project ID.
+
+5) In the [pom.xml](pom.xml) file, you should see the following code block:
+    ```$xslt
+    <plugin>
+        <groupId>com.google.cloud.tools</groupId>
+        <artifactId>appengine-maven-plugin</artifactId>
+        <version>2.2.0</version>
+    </plugin>
+    ```
+   Replace it with this:
+   ```$xslt
+   <plugin>
+       <groupId>com.google.cloud.tools</groupId>
+       <artifactId>appengine-maven-plugin</artifactId>
+       <configuration>
+           <deploy.version>1</deploy.version>
+           <deploy.projectId>YOUR_PROJECT_ID</deploy.projectId>
+       </configuration>
+       <version>2.2.0</version>
+   </plugin>
+   ```
+   Replace `YOUR_PROJECT_ID` with your project ID.
+
+### Enable APIs
+To enable an API, navigate to "APIs & Services" -> "Library" -> "<The API>" and press "Enable".
+Complete these steps for the following APIs:
+
+1) Gmail API
+2) Tasks API
+3) Google Calendar API
+4) Geocoding API
+5) Places API
+6) Directions API
+7) Secret Manager API
+
+### API Key Generation
+1) In the `src/main/resources` directory, create a new file called `KEYS.json`. It should have the following format:
+    ```$xslt
+    {
+      "apiKey" : "YOUR KEY",
+      "clientId" : "YOUR CLIENT ID"
+    }
+    ```
+
+2) In the [Google Cloud Console](https://console.cloud.google.com), navigate to "APIs and Services" -> "Credentials".
+
+3) Click "Create Credentials", and first select "OAuth Client ID".
+
+4) Select your Application Type as "Web Application", and select a name for the Client ID (anything is fine).
+
+5) Add any origins you may need to the "Authorized JavaScript Origins" and "Authorized redirect URIs".
+Add every URI that you anticipate using the application from (e.g. localhost:8080 (default localhost for this
+project), YOUR_PROJECT_ID.appspot.com (the appengine domain), etc.)
+
+6) Copy your project ID and place it in the pom.xml file.
+
+7) Click "Create Credentials" again and select "API Key".
+
+8) Copy the API Key into the pom.xml file.
+
+### Set up the GCloud SDK (if you're new to using the GCloud SDK)
+1) Run `gcloud init` and configure GCloud on your computer. When prompted, please enter the numeric choice that
+corresponds to the project ID you just created.
+
+2) Run `gcloud app create` to create an appengine project. Select any region you'd like, though it is suggested to
+select a region close to you. For more information, check this [link](https://cloud.google.com/appengine/docs/locations).
+
+### Running Application
+1) Run `mvn clean install` in the project root directory
+
+2) Run `appengine:deploy` to deploy the website
+
+3) Navigate to "YOUR_PROJECT_ID.appspot.com". You should see a deployed version of Blueprint!
```

Comment:
Interestingly enough, That above format still works. I'll also mention the other format

Reply:
I didn't know it was optional actually, but they apparently plan on making it mandatory: 

_Including REGION_ID.r in App Engine URLs is optional for existing apps and will soon be required for all new apps.
To ensure a smooth transition, we are slowly updating App Engine to use region IDs. If we haven't updated your Google Cloud project yet, you won't see a region ID for your app. Since the ID is optional for existing apps, you don't need to update URLs or make other changes once the region ID is available for your existing apps._